app/models/action.rb

```diff
@@ -3,5 +3,13 @@ class Action < ActiveRecord::Base
   belongs_to :member
 
   has_paper_trail on: [:update, :destroy]
+
+  after_create :update_member_donor_status
```

Comment:
I'll add my 2¢ to the conversation. As a general rule, I'd say ActiveRecord callbacks shouldn't create side effects _outside_ of their domain (preferably the record itself, but the collection if that's the case). Obviously you _can_, but it increases complexity in terms of how you're managing state.

Whilst going through the code earlier today, I realised that using a callback on the actions model is perhaps the easiest way to accomplish this due to accepting payments via GoCardless and BrainTree (and possibly more in the future as you've pointed out). However, orchestrating state changes in various resources is the kind of thing that belongs in a service. I think that's highlighted by the fact that you're relying on `form_data['is_subscription']` (part of the request?) inside an ActiveRecord model. That seems to belong in a controller, or a service (handling any type of request after all, be it `application/x-www-form-urlencoded` or `application/json`?)

If the final decision however is to use AR callbacks, I'd suggest replacing it for:

```rb
after_commit :update_member_donor_status, on: :create
```